src/NServiceBus.Core.Tests/Msmq/MsmqSubscriptionStorageQueueTests.cs

```diff
@@ -168,21 +250,34 @@ class OtherMessage : IMessage
 
         class FakeStorageQueue : IMsmqSubscriptionStorageQueue
         {
-            readonly List<Message> q = new List<Message>();
+            public readonly List<MsmqSubscriptionMessage> Messages = new List<MsmqSubscriptionMessage>();
 
-            public IEnumerable<Message> GetAllMessages()
+            public IEnumerable<MsmqSubscriptionMessage> GetAllMessages()
             {
-                return q.ToArray();
+                return Messages.ToArray();
             }
 
-            public void Send(Message toSend)
+            public string Send(string body, string label)
             {
-                q.Add(toSend);
+                // Wait for a short period of time to ensure multiple sends have different DateTime.Now values
+                // This is needed to ensure the expected order of messages when subscribing.
+                Thread.Sleep(1);
+                var id = Guid.NewGuid().ToString();
+
+                Messages.Add(new MsmqSubscriptionMessage
+                {
+                    ArrivedTime = DateTime.Now,
```

Comment:
I haven't fully review it yet. But how about using a fixed `DateTime `and then increment it by using `AddSeconds `or something like that instead of `Thread.Sleep.`